data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceTest.java

```diff
@@ -0,0 +1,116 @@
+    private ThresholdOptions thresholdOptions;
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+    private AwsCredentialsProvider awsCredentialsProvider;
+    private BucketOptions bucketOptions;
+    private ObjectKeyOptions objectKeyOptions;
+
+    @BeforeEach
+    void setUp() throws Exception {
+
+        s3SinkConfig = mock(S3SinkConfig.class);
+        thresholdOptions = mock(ThresholdOptions.class);
+        bucketOptions = mock(BucketOptions.class);
+        objectKeyOptions = mock(ObjectKeyOptions.class);
+        awsAuthenticationOptions = mock(AwsAuthenticationOptions.class);
+
+        when(objectKeyOptions.getNamePattern()).thenReturn("my-elb-%{yyyy-MM-dd'T'hh-mm-ss}");
+        when(s3SinkConfig.getThresholdOptions()).thenReturn(thresholdOptions);
+        when(s3SinkConfig.getThresholdOptions().getEventCount()).thenReturn(100);
+        when(s3SinkConfig.getThresholdOptions().getMaximumSize()).thenReturn(ByteCount.parse("1kb"));
+        when(s3SinkConfig.getThresholdOptions().getEventCollect()).thenReturn(Duration.ofSeconds(5));
+        when(s3SinkConfig.getBufferType()).thenReturn(BufferTypeOptions.LOCALFILE);
+
+        when(s3SinkConfig.getBucketOptions()).thenReturn(bucketOptions);
+        when(s3SinkConfig.getBucketOptions().getObjectKeyOptions()).thenReturn(objectKeyOptions);
+        when(s3SinkConfig.getBucketOptions().getBucketName()).thenReturn("dataprepper");
+        when(s3SinkConfig.getBucketOptions().getObjectKeyOptions().getPathPrefix()).thenReturn("logdata/");
+
+        when(s3SinkConfig.getAwsAuthenticationOptions()).thenReturn(awsAuthenticationOptions);
+        when(awsAuthenticationOptions.getAwsRegion()).thenReturn(Region.of("us-east-1"));
+        when(awsAuthenticationOptions.authenticateAwsConfiguration()).thenReturn(awsCredentialsProvider);
+    }
+
+    @Test
+    void test_call_createS3Client() {
+        S3SinkService s3SinkService = new S3SinkService(s3SinkConfig);
+        S3Client s3Client = s3SinkService.createS3Client();
+        assertNotNull(s3Client);
+    }
+
+    @Test
+    void test_call_processRecords() {
+        S3SinkService s3SinkService = new S3SinkService(s3SinkConfig);
+        assertNotNull(s3SinkService);
+        s3SinkService.processRecords(generateRandomStringEventRecord());
+        s3SinkService = mock(S3SinkService.class);
+        Collection<Record<Event>> records = generateRandomStringEventRecord();
+        s3SinkService.processRecords(records);
+        verify(s3SinkService, atLeastOnce()).processRecords(records);
+    }
+
+    @Test
+    void test_call_bufferAccumulator() {
+        Collection<Record<Event>> records = generateRandomStringEventRecord();
+        S3SinkService service = new S3SinkService(s3SinkConfig);
+        S3SinkWorker worker = mock(S3SinkWorker.class);
+        service.processRecords(records);
+        service.accumulateBufferEvents(worker);
+        verify(worker, only()).bufferAccumulator(any(BlockingQueue.class));

```

Comment:
Same here. You can check to the eventQueue is empty after doing `accumulateBufferEvents`

Reply:
These changes will be addressed as incremental changes, and they will be available in a separate PR.